testing/matrix-sdk-integration-testing/src/helpers.rs

```diff
@@ -212,19 +215,50 @@ impl Deref for SyncTokenAwareClient {
     }
 }
 
+/// Repeatedly run the given callback, at increasing intervals, until it returns
+/// `Some`, or the given timeout has elapsed.
+///
+/// The callback `cb` is called with the attempt number as an argument.
+///
+/// # Returns
+///
+/// The result of `cb` if it returned `Some`. If the timeout elapses, an error
+/// of type [`ElapsedError`].
+pub async fn wait_until_some<F, Fut, R>(
+    mut cb: F,
+    timeout_duration: Duration,
+) -> Result<R, ElapsedError>
+where
+    F: FnMut(u32) -> Fut,
+    Fut: Future<Output = Option<R>>,
+{
+    let start = Instant::now();
```

Comment:
I know that this is trivial but perhaps we could have used `backon` for this.

The body could also handle a bit of whitespace.

That being said, I'm not requesting anything to be changed.

Reply:
eh, yes, probably, on both fronts.